### PR TITLE
Add BBLFSH_ENDPOINT and BBLFSH_ENDPOINT environment variables

### PR DESCRIPTION
**Pull request by zurk**

### CODE REVIEW

File: ast2vec/__main__.py [hunk 1]
```diff
@@ -65,9 +65,10 @@ def get_parser() -> argparse.ArgumentParser:
 
     bblfsh_args = argparse.ArgumentParser(add_help=False)
     bblfsh_args.add_argument(
-        "--bblfsh", help="Babelfish server's endpoint, e.g. 0.0.0.0:9432.", dest="bblfsh_endpoint")
+        "--bblfsh", default=None, dest="bblfsh_endpoint",
+        help="Babelfish server's endpoint, e.g. 0.0.0.0:9432.")
     bblfsh_args.add_argument(
-        "--timeout", type=int, default=Repo2Base.DEFAULT_BBLFSH_TIMEOUT,
```
Comment (vmarkovtsev) [inline ast2vec/__main__.py]: This shows the hint in `--help`, please leave it

File: ast2vec/repo2/base.py [hunk 2]
```diff
@@ -495,21 +496,54 @@ def result_to_model_kwargs(self, result, url_or_path: str) -> dict:
...
+    if "BBLFSH_TIMEOUT" in os.environ:
+        try:
+            timeout = int(os.environ["BBLFSH_TIMEOUT"])
+            log.info("Get bblfsh timeout from BBLFSH_TIMEOUT environment variable: %s sec",
+                     timeout)
+            return timeout
+        except ValueError as e:
+            log.warning(("You provide wrong value: {} in BBLFSH_TIMEOUT environment variable. "
+                        "Should be integer").format(os.environ["BBLFSH_TIMEOUT"]))
+    log.info("You don't provide bblfsh timeout directly or in BBLFSH_TIMEOUT environment "
```
Comment (vmarkovtsev) [inline ast2vec/repo2/base.py]: DEBUG

You did not provide

File: ast2vec/repo2/base.py [hunk 3]
```diff
@@ -495,21 +496,54 @@ def result_to_model_kwargs(self, result, url_or_path: str) -> dict:
...
+        return bblfsh_timeout
+    log = logging.getLogger("bblfsh")
+    if "BBLFSH_TIMEOUT" in os.environ:
+        try:
+            timeout = int(os.environ["BBLFSH_TIMEOUT"])
+            log.info("Get bblfsh timeout from BBLFSH_TIMEOUT environment variable: %s sec",
+                     timeout)
+            return timeout
+        except ValueError as e:
+            log.warning(("You provide wrong value: {} in BBLFSH_TIMEOUT environment variable. "
```
Comment (vmarkovtsev) [inline ast2vec/repo2/base.py]: Please use the built-in logging syntax with %s
log.warning("You provide**d** wrong value: %s", env_timeout)

File: ast2vec/repo2/base.py [hunk 4]
```diff
@@ -495,21 +496,54 @@ def result_to_model_kwargs(self, result, url_or_path: str) -> dict:
         raise NotImplementedError
 
 
-def ensure_bblfsh_is_running_noexc():
+def resolve_bblfsh_timeout(bblfsh_timeout):
+    if bblfsh_timeout is not None:
+        return bblfsh_timeout
+    log = logging.getLogger("bblfsh")
+    if "BBLFSH_TIMEOUT" in os.environ:
```
Comment (vmarkovtsev) [inline ast2vec/repo2/base.py]:
```
env_timeout = os.getenv("BBLFSH_TIMEOUT")
if env_timeout:
    ...
```

File: ast2vec/repo2/base.py [hunk 5]
```diff
@@ -495,21 +496,54 @@ def result_to_model_kwargs(self, result, url_or_path: str) -> dict:
...
+                        "Should be integer").format(os.environ["BBLFSH_TIMEOUT"]))
+    log.info("You don't provide bblfsh timeout directly or in BBLFSH_TIMEOUT environment "
+             "variable. Default %s sec will be used", DEFAULT_BBLFSH_TIMEOUT)
+    return DEFAULT_BBLFSH_TIMEOUT
+
+
+def resolve_bblfsh_endpoint(bblfsh_endpoint):
+    if bblfsh_endpoint is not None:
+        return bblfsh_endpoint
+    log = logging.getLogger("bblfsh")
```
Comment (vmarkovtsev) [inline ast2vec/repo2/base.py]: `os.getenv()`

File: ast2vec/repo2/base.py [hunk 6]
```diff
@@ -495,21 +496,54 @@ def result_to_model_kwargs(self, result, url_or_path: str) -> dict:
...
+    if "BBLFSH_ENDPOINT" in os.environ:
+        log.info("Get bblfsh endpoint from BBLFSH_ENDPOINT environment variable: %s",
+                 os.environ["BBLFSH_ENDPOINT"])
+        return os.environ["BBLFSH_ENDPOINT"]
+    log.info("You don't provide bblfsh endpoint directly or in BBLFSH_ENDPOINT environment "
```
Comment (vmarkovtsev) [inline ast2vec/repo2/base.py]: DEBUG

You did not

File: ast2vec/repo2/base.py [hunk 7]
```diff
@@ -495,21 +496,54 @@ def result_to_model_kwargs(self, result, url_or_path: str) -> dict:
...
+             "variable. Default %s sec will be used", DEFAULT_BBLFSH_TIMEOUT)
+    return DEFAULT_BBLFSH_TIMEOUT
+
+
+def resolve_bblfsh_endpoint(bblfsh_endpoint):
+    if bblfsh_endpoint is not None:
+        return bblfsh_endpoint
+    log = logging.getLogger("bblfsh")
+    if "BBLFSH_ENDPOINT" in os.environ:
+        log.info("Get bblfsh endpoint from BBLFSH_ENDPOINT environment variable: %s",
```
Comment (vmarkovtsev) [inline ast2vec/repo2/base.py]: Got

File: ast2vec/repo2/base.py [hunk 8]
```diff
@@ -495,21 +496,54 @@ def result_to_model_kwargs(self, result, url_or_path: str) -> dict:
...
-    try:
-        ensure_bblfsh_is_running()
-    except:
-        log = logging.getLogger("bblfsh")
-        message = "Failed to ensure that the Babelfish server is running."
-        if log.isEnabledFor(logging.DEBUG):
-            log.exception(message)
-        else:
-            log.warning(message)
+    if resolve_bblfsh_endpoint(bblfsh_endpoint) == DEFAULT_BBLFSH_ENDPOINT:
```
Comment (vmarkovtsev) [inline ast2vec/repo2/base.py]:
```
if resolve_bblfsh_endpoint(bblfsh_endpoint) != DEFAULT_BBLFSH_ENDPOINT:
    return
```

File: ast2vec/__main__.py (hunk 1 above)
Comment (zurk) [reply ast2vec/__main__.py]: ok, I find how to fix it and will create another PR for __main__ to show default value.

**Pull request merged**